py4DSTEM/process/polar/polar_analysis.py

```diff
@@ -596,18 +600,298 @@ def calculate_FEM_local(
+            im_var = np.ma.median((im - im_mean) ** 2, axis=0)
+
+        self.local_radial_mean[rx, ry] = im_mean
+        self.local_radial_var[rx, ry] = im_var
+
+    if plot_normalized_variance:
+        fig, ax = plt.subplots(figsize=figsize)
+
+        sig = self.local_radial_var / self.local_radial_mean**2
+        if use_median:
+            sig_plot = np.median(sig, axis=(0, 1))
+        else:
+            sig_plot = np.mean(sig, axis=(0, 1))
+
+        ax.plot(
+            self.qq,
+            sig_plot,
+        )
+        ax.set_xlabel(
+            "Scattering Vector (" + self.calibration.get_Q_pixel_units() + ")"
+        )
+        ax.set_ylabel("Normalized Variance")
+        ax.set_xlim((self.qq[0], self.qq[-1]))
+
+        # self.radial_all_std[rx, ry] = np.sqrt(
+        #     np.mean((self.data[rx, ry] - self.radial_all[rx, ry][None]) ** 2, axis=0)
+        # )
+
+    if return_values:
+        if returnfig:
+            return self.local_radial_mean, self.local_radial_var, fig, ax
+        else:
+            return self.local_radial_mean, self.local_radial_var
+    else:
+        if returnfig:
+            return fig, ax
+
+
+def calculate_annular_symmetry(
+    self,
+    max_symmetry=12,
+    mask_realspace=None,
+    plot_result=False,
+    figsize=(8, 4),
+    returnval=False,
+    progress_bar=True,
+):
     """
+    This function calculates radial symmetry of diffraction patterns, typically applied
+    to amorphous scattering, but it can also be used for crystalline Bragg diffraction.
 
-    pass
+    Parameters
+    --------
+    self: PolarDatacube
+        Polar transformation datacube
+    max_symmetry: int
+        Symmetry orders will be computed from 1 to max_symmetry for n-fold symmetry orders.
+    mask_realspace: np.array
+        Boolean mask, symmetries will only be computed at probe positions wheremask is True.
+    plot_result: bool
+        Plot the resulting array
+    figsize: (float, float)
+        Size of the plot.
+    returnval: bool
+        Set to true to return the symmetry array.
+    progress_bar: bool
+        Show progress bar during calculation.
+
+    Returns
+    --------
+    annular_symmetry: np.array
+        Array with annular symmetry magnitudes, with shape [max_symmetry, num_radial_bins]
+
+    """
+
+    # Initialize outputs
+    self.annular_symmetry_max = max_symmetry
+    self.annular_symmetry = np.zeros(
+        (
+            self.data_raw.shape[0],
+            self.data_raw.shape[1],
+            max_symmetry,
+            self.polar_shape[1],
+        )
+    )
+
+    # Loop over all probe positions
+    for rx, ry in tqdmnd(
+        self._datacube.shape[0],
+        self._datacube.shape[1],
+        desc="Annular symmetry",
+        unit=" probe positions",
+        disable=not progress_bar,
+    ):
+        # Get polar transformed image
+        im = self.transform(
+            self.data_raw.data[rx, ry],
+            # returnval = 'zeros',
+        )
```

Comment:
Is there a reason to repeat the transformation here, since this could operate on a PolarDatacube that would already be transformed once ahead of time?

Reply:
PolarDataCube currently generates the images "on demand" rather than precomputing them.